### PR TITLE
SOLR-6613: TextField.analyzeMultiTerm does not throw an exception…

### PR DESCRIPTION
**Pull request by bruno-roustant**
when Analyzer returns no term.

### CODE REVIEW

File: solr/CHANGES.txt [hunk 1]
```diff
@@ -212,6 +212,8 @@ Bug Fixes
 
 * SOLR-14122: SimUtils converts v2 to v1 request params incorrectly. (Li Cao, ab)
 
+* SOLR-6613: TextField.analyzeMultiTerm does not throw an exception when Analyzer returns no term. (Bruno Roustant)
```
Comment (dsmiley) [inline solr/CHANGES.txt]: "no term" -> "no terms"

File: solr/core/src/java/org/apache/solr/schema/TextField.java [hunk 2]
```diff
@@ -173,8 +173,10 @@ public static BytesRef analyzeMultiTerm(String field, String part, Analyzer anal
 
       TermToBytesRefAttribute termAtt = source.getAttribute(TermToBytesRefAttribute.class);
 
-      if (!source.incrementToken())
-        throw  new SolrException(SolrException.ErrorCode.BAD_REQUEST,"analyzer returned no terms for multiTerm term: " + part);
+      if (!source.incrementToken()) {
+        // Accept no token because it may have been filtered out by a StopFilter for example.
+        return null;
+      }
```
Comment (dsmiley) [inline solr/core/src/java/org/apache/solr/schema/TextField.java]: Optional: Consider adding javadocs for TextField.analyzeMultiTerm().  Yes, someone else should have long ago

File: solr/core/src/test/org/apache/solr/schema/TestTextField.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    assertNull(termBytes);
+
+    // One term provided by the WhitespaceTokenizer for the multi-term part.
+    // This is the regular case. Check TextField.analyzeMultiTerm returns it (and does not throw an exception).
+    termBytes = TextField.analyzeMultiTerm("field", "Sol", new WhitespaceAnalyzer());
+    assertEquals("Sol", termBytes.utf8ToString());
+
+    // Two terms provided by the WhitespaceTokenizer for the multi-term part.
+    // This is not allowed. Expect an exception.
+    try {
```
Comment (dsmiley) [inline solr/core/src/test/org/apache/solr/schema/TestTextField.java]: See expectThrows() in-scope, instead.

File: solr/CHANGES.txt (hunk 1 above)
Comment (bruno-roustant) [reply solr/CHANGES.txt]: Hum right, this strange rule of negation and plural. There is the same notion in French and I don't like it neither. To me it means "not a single term" or "zero term" so why should we put plural.
But ok I'll change that.

File: solr/core/src/test/org/apache/solr/schema/TestTextField.java (hunk 3 above)
Comment (bruno-roustant) [reply solr/core/src/test/org/apache/solr/schema/TestTextField.java]: Thanks

**Pull request closed without merging**